src/chainstate/burn/db/sortdb.rs

```diff
@@ -681,17 +679,12 @@ impl <'a> SortitionHandleTx <'a> {
                 if reward_set_size == 0 {
                     Ok(None)
                 } else {
-                    let chosen_recipients = reward_set_vrf_seed.choose(OUTPUTS_PER_COMMIT.try_into().expect("BUG: u32 overflow in PoX outputs per commit"),
-                                                                       reward_set_size as u32);
-                    let mut recipients = vec![];
-                    for ix in chosen_recipients.into_iter() {
-                        let ix = u16::try_from(ix).unwrap();
-                        let recipient = self.get_reward_set_entry(ix)?;
-                        recipients.push((recipient, ix));
-                    }
+                    let chosen_recipient = reward_set_vrf_seed.choose(reward_set_size as u32);
```

Comment:
Should we add a runtime `assert!` here to ensure that `chosen_recipient` is in range `[0, reward_set_size)`?

Reply:
If we wanted an `assert!`, I'd rather see that in the implementation of `choose`, because that's where the behavior is defined.

Reply:
Fine by me.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1863/commits/9238aa1032bbb2cea8f5fc1366e7d912b35cbdb7